scripts/release.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+set -e
+
+IS_PRERELEASE=true
+
+if [[ ! $- == *i* ]]
+then
+    echo "Warning: this is an interactive script"
+fi
+
+if [[ -z $CIRCLE_CI_TOKEN ]]
+then
+  echo "Please provide a CircleCI API token in \$CIRCLE_CI_TOKEN"
+  exit 1
+fi
+
+if [[ -z $GITHUB_TOKEN ]]
+then
+  echo "Please provide a Github personal access token in \$GITHUB_TOKEN"
+  exit 1
+fi
+
+if [[ -z $GOPATH ]]
+then
+  echo "Please install Golang"
+  exit 1
+fi
+
+if [[ $(ls $GOPATH/bin/hub | wc -l) == "0" ]]
+then
+  echo "Installing required tool Github 'hub'"
+  go get github.com/github/hub
+fi
+
+read -p "Creating a release off \"$(git branch | grep \* | cut -d ' ' -f2)\", is this correct? [Y/n] "  Y
+if [[ -n $Y && "$(echo $Y | tr '[:upper:]' '[:lower:]')" != "y" ]]
+then
+  exit 0
+fi
+
+read -p "Enter release version (eg v1.2.3 or test-myfeature) " VERSION
+echo "Make sure that you can sign commits on this machine with your GPG key: "
+echo "https://help.github.com/articles/signing-commits/"
+read -p "Press enter to edit git tag description " Y
+
+git tag -s $VERSION
```

Comment:
If things fail further down (eg: transient error), can these git commands be re-run with the same version?
Would some manual cleanup be required before running it again, eg: deleting the tag locally and on the remote)

Reply:
Totally non-idempotent script!

(I added a check for if the tag exists on the remote, followed by a prompt)